core/src/main/java/org/apache/iceberg/CatalogUtil.java

```diff
@@ -137,6 +138,18 @@ public static void dropTableData(FileIO io, TableMetadata metadata) {
     deleteFile(io, metadata.metadataFileLocation(), "metadata");
   }
 
+  /**
+   * Drops view metadata files referenced by ViewMetadata.
+   *
+   * <p>This should be called by dropView implementations
+   *
+   * @param io a FileIO to use for deletes
+   * @param metadata the last valid ViewMetadata instance for a dropped view.
+   */
+  public static void dropViewMetadata(FileIO io, ViewMetadata metadata) {
+    deleteFile(io, metadata.metadataFileLocation(), "metadata");
```

Comment:
Shouldn't this also respect the `gcEnabled` flag like drop table?

Reply:
@danielcweeks it looks like for the table case we don't respect the gc flag when dropping the metadata file itself: https://github.com/apache/iceberg/blob/bd35447383f3aafa51436f0d45e15d240cfecb47/core/src/main/java/org/apache/iceberg/CatalogUtil.java#L138
I'm ok either way but I think we should be consistent with how we do it for tables (aka not respecting the gc flag when deleting view metadata)

Reply:
actually wrt https://github.com/apache/iceberg/pull/9852/files#r1757411419 it's probably better to delete view metadata only when GC is enabled